content/docs/2.0/scalers/aws-cloudwatch.md

```diff
@@ -26,9 +26,9 @@ triggers:
     # Required: region
     awsRegion: "eu-west-1"
     # Optional: AWS Access Key ID, can use TriggerAuthentication as well
-    awsAccessKeyID: AWS_ACCESS_KEY_ID # default AWS_ACCESS_KEY_ID
+    awsAccessKeyIDFromEnv: AWS_ACCESS_KEY_ID # default AWS_ACCESS_KEY_ID
```

Comment:
Can't we specify literal values of the ID anymore? Think that's still valuable.

Reply:
Looks like the code still supports this so maybe just good to list both options every time?

Reply:
Yes, all the properties with `FromEnv` suffix can be overridden by one that doesn't have `FromEnv`. I was thinking I can add a `> ` note on all the scalers with something like 

> Metadata properties with `*FromEnv` suffix are resolved from the environment of the pods in scaleTargetRef. It's recommended to source these settings from a Secret on the `scaleTargetRef` or through a TriggerAuthentication CRD. Dropping `FromEnv` will use the literal value defined in the ScaledObject

Reply:
I'm a bit conflicted though about this not discouraging users enough from storing secrets in the ScaledObject CRD. What do you think about having a `*PlainText` suffix for ones with literal values in the CRD itself? I understand the ease of use when you're trying to get started by defining everything in one CRD, but just want a clear indication that there is a better way of doing it.

Reply:
There are a few things:
1. I would simply add every property to our trigger spec & parameter list in the docs, this gives a clear overview of the configuration options
2. A notice on how it works is definitely a good idea, certainly if one will be favored over another
3. I do get your point about stimulating them to use secrets for secrets, I don't think we should have plain text options for all the options. For example, literal value for `awsAccessKeyID` is fine and we can just document it. However, `awsSecretAccessKey` should only be configurable via `awsSecretAccessKeyFromEnv` or through TriggerAuthentication (and thus documented like this)

That way; safe things can be configured as literal values, env var or TriggerAuth while secrets can only be configured through env var or Trigger Auth.

I personally think that's the sweet spot.

Reply:
Adding @zroubalik as he might have some input as well

Reply:
Agree with @tomkerkhove

Reply:
[This](https://github.com/kedacore/keda/pull/1131) makes the change needed to resolve secrets only for properties using `FromEnv`. Updating the docs now